x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/v2/TransportRollupAction.java

```diff
@@ -218,46 +219,45 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
             }
 
             @Override
-            public ClusterState execute(ClusterState currentState) {
+            public ClusterState execute(ClusterState currentState) throws IOException {
                 IndexMetadata rollupIndexMetadata = currentState.getMetadata().index(rollupIndexName);
                 Index rollupIndex = rollupIndexMetadata.getIndex();
                 Map<String, String> idxMetadata = currentState.getMetadata().index(originalIndexName)
-                    .getCustomData(RollupMetadata.TYPE);
-                String rollupGroupKeyName = (idxMetadata == null) ?
-                    originalIndexName : idxMetadata.get(RollupMetadata.SOURCE_INDEX_NAME_META_FIELD);
+                    .getCustomData(RollupIndexMetadata.TYPE);
+
+                // Add the source index name to the rollup index metadata. If the original index is a rollup index itself
+                // we will add the name of the raw index that we initially rolled up.
+                String rollupSourceIndexName = idxMetadata != null ?
+                    idxMetadata.get(RollupIndexMetadata.SOURCE_INDEX_NAME_META_FIELD) : originalIndexName;
                 Map<String, String> rollupIndexRollupMetadata = new HashMap<>();
-                rollupIndexRollupMetadata.put(RollupMetadata.SOURCE_INDEX_NAME_META_FIELD, rollupGroupKeyName);
-                final RollupMetadata rollupMetadata = currentState.metadata().custom(RollupMetadata.TYPE);
-                final Map<String, RollupGroup> rollupGroups;
-                if (rollupMetadata == null) {
-                    rollupGroups = new HashMap<>();
-                } else {
-                    rollupGroups = new HashMap<>(rollupMetadata.rollupGroups());
-                }
+                rollupIndexRollupMetadata.put(RollupIndexMetadata.SOURCE_INDEX_NAME_META_FIELD, rollupSourceIndexName);
+                // Add metadata about the rollup configuration
```

Comment:
That should be enough for now. The rest of the information will be recorded in the mapping so no need to introduce the rollup metadata object below.